Source/REFACTORING/Collection/Delegate/BaseCollectionDelegate.swift

```diff
@@ -6,29 +6,23 @@
 //  Copyright © 2021 Александр Кравченков. All rights reserved.
 //
 
-import Foundation
-
-protocol CollectionDelegate: UICollectionViewDelegate {
-    var manager: BaseCollectionManager? { get set }
-    var collectionPlugins: PluginCollection<BaseCollectionPlugin<CollectionEvent>> { get set }
-    var scrollPlugins: PluginCollection<BaseCollectionPlugin<ScrollEvent>> { get set }
-}
+import UIKit
 
 // Base implementation for UICollectionViewDelegate protocol. Use it if NO special logic required.
-open class BaseCollectionDelegate: NSObject {
+open class BaseCollectionDelegate: NSObject, CollectionDelegate {
 
     // MARK: - Properties
 
-    weak var manager: BaseCollectionManager?
+    weak public var manager: BaseCollectionManager?
 
-    var collectionPlugins = PluginCollection<BaseCollectionPlugin<CollectionEvent>>()
-    var scrollPlugins = PluginCollection<BaseCollectionPlugin<ScrollEvent>>()
+    public var collectionPlugins = PluginCollection<BaseCollectionPlugin<CollectionEvent>>()
+    public var scrollPlugins = PluginCollection<BaseCollectionPlugin<ScrollEvent>>()
 
 }
 
 // MARK: - UICollectionViewDelegate
 
-extension BaseCollectionDelegate: CollectionDelegate {
+extension BaseCollectionDelegate: UICollectionViewDelegate {
```

Comment:
```suggestion
extension BaseCollectionDelegate {
```

`CollectionDelegate` уже подписан на `UICollectionViewDelegate`